OPHD/States/PlanetSelectState.cpp

```diff
@@ -135,10 +129,10 @@ State* PlanetSelectState::update()
 	{
 		return this;
 	}
-	else if (planetSelection != planetSelectionInvalid)
+	else if (mPlanetSelection != constants::NO_SELECTION)
```

Comment:
I would caution against re-using `constants::NO_SELECTION` in many different contexts. Particularly when the objects being selected are of widely varying types.

Sentinel values are often context dependent. For instance, array based sentinel values typically use `-1`, while pointer based sentinel values typically use `0` or `nullptr`. If a data type changes between array based or pointer based, the sentinel value may also need to be updated. Keeping a local name for the sentinel values makes such updates easier, as you don't need to sort out which uses of the sentinel name apply to that particular collection type.

I have seen some rather unfortunate cases where some collections end up with effectively two sentinel values, depending on context, because this sort of things wasn't handled in a consistent manner, and it led to some pretty unusual bugs. In particular OP2, and the garage save/load crash bug.